common/thanos/templates/alerts/_thanos-store.alerts.tpl

```diff
@@ -4,29 +4,82 @@ groups:
 - name: thanos-store.alerts
   rules:
     - alert: ThanosStoreGrpcErrorRate
-      expr: rate(grpc_server_handled_total{grpc_code=~"Unknown|ResourceExhausted|Internal|Unavailable",app="thanos-store",prometheus="{{ include "thanos.name" . }}"}[5m]) > 0
+      expr: |
+        (
+          sum by (thanos) (rate(grpc_server_handled_total{grpc_code=~"Unknown|ResourceExhausted|Internal|Unavailable|DataLoss|DeadlineExceeded", job=~".*thanos.*store.*", thanos="{{ include "thanos.name" . }}"}[5m]))
+        /
+          sum by (thanos) (rate(grpc_server_started_total{job=~".*thanos.*store.*", thanos="{{ include "thanos.name" . }}"}[5m]))
+        * 100 > 5
+        )
       for: 5m
       labels:
-        context: thanos
         service: {{ default "metrics" $root.Values.alerts.service }}
-        support_group: observability
+        support_group: {{ default "observability" $root.Values.alerts.support_group }}
         severity: info
-        playbook: 'docs/support/playbook/prometheus/thanos_store.html'
-        meta: 'Thanos store is returning errors for Prometheus `{{`{{ $labels.prometheus }}`}}`'
+        meta: Thanos Store `{{`{{ $labels.thanos }}`}}` is failing to handle requests.
       annotations:
-        description: 'Thanos Store is returning Internal/Unavailable errors for Prometheus `{{`{{ $labels.prometheus }}`}}`. Long Term Storage Prometheus queries are failing.'
-        summary: Thanos store has errors
+        description: |
+          Thanos Store `{{`{{ $labels.thanos }}`}}` is failing
+          to handle `{{`{{ $value | humanize }}`}}%` of gRPC requests.
+        summary: Thanos Store is failing to handle gRPC requests.
 
-    - alert: ThanosStoreBucketOperationsFailed
-      expr: rate(thanos_objstore_bucket_operation_failures_total{app="thanos-store",prometheus="{{ include "thanos.name" . }}"}[5m]) > 0
-      for: 5m
+    - alert: ThanosStoreSeriesGateLatencyHigh
+      expr: |
+        (
+          histogram_quantile(0.99, sum by (thanos, le) (rate(thanos_bucket_store_series_gate_duration_seconds_bucket{job=~".*thanos.*store.*", thanos="{{ include "thanos.name" . }}"}[5m]))) > 2
+        and
+          sum by (thanos) (rate(thanos_bucket_store_series_gate_duration_seconds_count{job=~".*thanos.*store.*", thanos="{{ include "thanos.name" . }}"}[5m])) > 0
+        )
+      for: 10m
+      labels:
+        service: {{ default "metrics" $root.Values.alerts.service }}
+        support_group: {{ default "observability" $root.Values.alerts.support_group }}
+        severity: info
+        meta: Thanos Store `{{`{{ $labels.thanos }}`}}` has a 99th percentile latency for store series gate requests.
+      annotations:
+        description: |
+          Thanos Store `{{`{{ $labels.thanos }}`}}` has a 99th 
+          percentile latency of `{{`{{ $value }}`}}`seconds 
+          for store series gate requests. 
```

Comment:
Would be interesting to know what `series gate requests` are about. Guess we have to figure this out, once it happens. Hence `info` is good 👍